src/ripple/net/impl/RPCCall.cpp

```diff
@@ -482,17 +482,31 @@ class RPCParser
         return jvRequest;
     }
 
-    // connect <ip> [port]
+    // connect <ip[:port]> [port]
     Json::Value
     parseConnect(Json::Value const& jvParams)
     {
         Json::Value jvRequest(Json::objectValue);
-
-        jvRequest[jss::ip] = jvParams[0u].asString();
-
+        std::string ip = jvParams[0u].asString();
         if (jvParams.size() == 2)
+        {
+            jvRequest[jss::ip] = ip;
             jvRequest[jss::port] = jvParams[1u].asUInt();
+            return jvRequest;
+        }
+
+        // handle case where there is one argument of the form ip:port
+        std::size_t colon = ip.find_last_of(":");
+        if (colon != std::string::npos)
+        {
+            jvRequest[jss::ip] = std::string{ip, 0, colon};
+            jvRequest[jss::port] =
+                Json::Value{std::string{ip, colon + 1}}.asUInt();
+            return jvRequest;
+        }
```

Comment:
This logic breaks if you specify an IPv6 address, whether you use brackets or not, and that can be a moderately annoying problem.

Perhaps considering checking if `ip` can be successfully parsed as either an IPv6 addresses using, for example,
[make_address_v6](https://www.boost.org/doc/libs/1_80_0/doc/html/boost_asio/reference/ip__address_v6/make_address_v6.html) and only doing this is if it isn't.